R/RunQuery.R

```diff
@@ -46,7 +55,25 @@ RunQuery <- function(strQuery, df) {
     cli::cli_text("Creating a new temporary DuckDB connection.")
     con <- DBI::dbConnect(duckdb::duckdb())
     temp_table_name <- paste0("temp_table_", format(Sys.time(), "%Y%m%d_%H%M%S"))
-    DBI::dbWriteTable(con, temp_table_name, df)
+    append_tab = FALSE
+    if (bUseSchema) {
+      create_tab_query <- lColumnMapping %>%
+        map_chr(function(mapping) {
+          type <- case_when(mapping$type == "Date" ~ "DATE",
+                            mapping$type == "numeric" ~ "DOUBLE",
+                            mapping$type == "integer" ~ "INTEGER",
+                            mapping$type == "character" ~ "VARCHAR",
+                            T ~ "VARCHAR")
+            glue("{mapping$source} {type}")
+        }) %>%
+        paste(collapse = ", ")
+      create_tab_query <- glue("CREATE TABLE {temp_table_name} ({create_tab_query})")
+      dbExecute(con, create_tab_query)
+      # set up arguments for dbWriteTable
+      append_tab = TRUE
+      df <- df %>% select(sapply(lColumnMapping, function(x) x$source) %>% as.vector())
```

Comment:
yeah- `map_chr` is a better solution.

Reply:
made an edit, but still needed to unname() the named vector because it select was grabbing the names, which is the target column, and we need the source column, but its a bit clearer what is going on now